### PR TITLE
feat(Typechecking): added type definition for TrackingPoint and fixed…

### PR DESCRIPTION
**Pull request by pimdewit**
… type expectations for methods

### CODE REVIEW

File: src/utils.js [hunk 1]
```diff
@@ -79,14 +79,14 @@ export function booleanGetter(element, attributeName) {
...
-  element.setAttribute(attributeName, value);
+  element.setAttribute(attributeName, `${value}`);
 }
 
 /**
  * Standard getter for a Custom Element int property reflected to attribute.
  * @param {HTMLElement} element
  * @param {string} attributeName
- * @param {string} [defaultValue=0]
+ * @param {number|string} [defaultValue=0]
```
Comment (ciampo) [inline src/utils.js]: Do you think this should be always a number actually?

File: src/utils.js [hunk 2]
```diff
@@ -151,7 +151,7 @@ export function getCSSCustomProperty(element, propertyName) {
  * @param {string|number} propertyValue The property value.
  */
 export function setCSSCustomProperty(element, propertyName, propertyValue) {
-  element.style.setProperty(propertyName, propertyValue);
+  element.style.setProperty(propertyName, `${propertyValue}`);
```
Comment (ciampo) [inline src/utils.js]: Any particular reason to force it to be a string?

File: src/macro-carousel/macro-carousel.js [hunk 3]
```diff
@@ -1328,7 +1334,7 @@ value. Add CSS units to its value to avoid breaking the slides layout.`);
     this._slides = this._getSlides();
     this._slides.forEach(slide => {
       if (!slide.element.hasAttribute(ATTRS.STANDARD.TABINDEX)) {
-        slide.element.setAttribute(ATTRS.STANDARD.TABINDEX, -1);
+        slide.element.setAttribute(ATTRS.STANDARD.TABINDEX, '-1');
```
Comment (ciampo) [inline src/macro-carousel/macro-carousel.js]: We should probably apply the same changes to other `setAttribute` calls in the project

File: src/macro-carousel/macro-carousel.js [hunk 4]
```diff
@@ -480,7 +486,7 @@ class MacroCarousel extends HTMLElement {
    * binding every function. See
    * https://medium.com/@WebReflection/dom-handleevent-a-cross-platform-standard-since-year-2000-5bf17287fd38
    *
-   * @param {Event} e Any event.
+   * @param {Event|KeyboardEvent} e Any event.
```
Comment (ciampo) [inline src/macro-carousel/macro-carousel.js]: I think `KeyboardEvent` is a subclass of `Event`, is there any specific reason why you want to specify it?

File: src/utils.js (hunk 2 above)
Comment (pimdewit) [reply src/utils.js]: setProperty expects a string https://developer.mozilla.org/en-US/docs/Web/API/CSSStyleDeclaration/setProperty

File: src/utils.js (hunk 1 above)
Comment (pimdewit) [reply src/utils.js]: I do yeah

File: src/macro-carousel/macro-carousel.js (hunk 4 above)
Comment (pimdewit) [reply src/macro-carousel/macro-carousel.js]: Mainly to satisfy IDE's, not a biggie - will revert

File: src/macro-carousel/macro-carousel.js (hunk 3 above)
Comment (pimdewit) [reply src/macro-carousel/macro-carousel.js]: I can't spot any other occurrences where we have to do type coercion

**Pull request merged**